display_information/add-graphics-with-renderer/src/main/java/com/esri/samples/add_graphics_with_renderer/AddGraphicsWithRendererSample.java

```diff
@@ -108,18 +118,83 @@ public void start(Stage stage) {
       polygonGeometry.addPoint(-20e5, -20e5);
       Graphic polygonGraphic = new Graphic(polygonGeometry.toGeometry());
       polygonGraphicOverlay.getGraphics().add(polygonGraphic);
-      SimpleFillSymbol polygonSymbol = new SimpleFillSymbol(SimpleFillSymbol.Style.SOLID, 0xFFFFFF00, null);
+      SimpleFillSymbol polygonSymbol = new SimpleFillSymbol(SimpleFillSymbol.Style.SOLID, ColorUtil.colorToArgb(Color.YELLOW), null);
       SimpleRenderer polygonRenderer = new SimpleRenderer(polygonSymbol);
       polygonGraphicOverlay.setRenderer(polygonRenderer);
       mapView.getGraphicsOverlays().add(polygonGraphicOverlay);
 
+      // polygon with curve segments (red heart) graphic
+      GraphicsOverlay curvedGraphicOverlay = new GraphicsOverlay();
+      // create a simple fill symbol with outline
+      SimpleLineSymbol curvedLineSymbol = new SimpleLineSymbol(SimpleLineSymbol.Style.SOLID, ColorUtil.colorToArgb(Color.BLACK), 1);
+      SimpleFillSymbol curvedFillSymbol = new SimpleFillSymbol(SimpleFillSymbol.Style.SOLID, ColorUtil.colorToArgb(Color.RED), curvedLineSymbol);
+      SimpleRenderer curvedRenderer = new SimpleRenderer(curvedFillSymbol);
+      curvedGraphicOverlay.setRenderer(curvedRenderer);
+      // create a heart shaped graphic and add it to the map view
+      Point originPointForHeart = new Point(40e5, 5e5, SpatialReferences.getWebMercator());
+      Geometry heartGeometry = makeHeartGeometry(originPointForHeart);
+      Graphic heartGraphic = new Graphic(heartGeometry);
+      curvedGraphicOverlay.getGraphics().add(heartGraphic);
+      mapView.getGraphicsOverlays().add(curvedGraphicOverlay);
+
       // add the map view to stack pane
       stackPane.getChildren().add(mapView);
+
     } catch (Exception e) {
       e.printStackTrace();
     }
   }
 
+  /**
+   * Create a heart-shape geometry with bezier and elliptic arc segments.
+   *
+   * @param centerOfHeart the center of the square that contains the heart shape
+   * @return a heart-shape geometry
+   */
+  private Geometry makeHeartGeometry(Point centerOfHeart) {
```

Comment:
Does this need to be a separate method? It does remove some lines from the main method but maybe isn't necessary if the comments separate out each of the graphics quite clearly?

Reply:
I was back and forth with this too. I think in this case, even though it's single use, the method approach works well as it keeps the logic for setting the renderers across each different geometry standard and bite sized, e.g. create the graphics overlay, create the symbol and fill, create the renderer from that and apply to the graphics overlay, then add the graphic to the overlay. it's a good repeated code "story" for each geometry, and the user can dig down into the method to find more details for the heart shape. 

we will have more functionality coming for the curves API, I think it might be nice to have a separate sample for this so that the curve creation code is right at the centre (or the heart? :p ) of the sample.